### ISSUE TITLE
"Links to cross-origin destinations are unsafe" on <a> without href

### ISSUE THREAD

**Issue by Onderhond** [GoogleChrome/lighthouse#11647]
#### What is the current behavior?
Lighthouse gives an error on a link-tag without no href, stating "Links to cross-origin destinations are unsafe"

#### What is the expected behavior?
Lighthouse only reports on links that are actually cross-origin

#### Environment Information
* Affected Channels: DevTools
* Lighthouse version: 6.2.0
* Chrome version: 86
* Node.js version: //
* Operating System: Windows 10

**Comment by patrickhulce** [comment]
Thanks for filing @Onderhond! When this happens there should be a warning that says "Unable to determine the destination for anchor. If not used as a hyperlink, consider removing target=_blank."

I'm not able to reproduce this without the warning being present. If you are, would you be able to provide a URL example?

**Comment by Onderhond** [comment]
Replicated the bug (noticed it on ACC server, so can't give you an URL) and (kind of) found out what the problem was.

When I made a specific test case the error was still there, but the listed a tag pointed to a different link (that was in violation). On the ACC, it still points to a link without href (active state of a language navigation). When the site goes in production (later this month) and the problem persists, I'll add the url. 

But that changes the bug behavior to **link on violating a-tag not correct**

**Comment by patrickhulce** [comment]
Alright thanks @Onderhond! We'll wait for the public URL.

**Comment by Onderhond** [comment]
Not sure what the deal is with Lighthouse, but it seems this bug keeps reporting differently.

I've got a live URL now though: https://www.uantwerpen.be/nl/

The error seems to be in the SEO tab now. There it says:
**Links are not crawlable**

It lists 2 instances. I admit the footer nav usage is a bit dubious, but the active state in the language navigation shouldn't throw this error. From what I've read, it is acceptable to use <a> without href to mark up an active state in a navigation.

**Comment by patrickhulce** [comment]
> From what I've read, it is acceptable to use without href to mark up an active state in a navigation.

If the page is actually linking to the current active state (example: the NL `<a>` actually links to `/nl`), then of course! In these instances though these text elements are not acting as links, so the audit is capturing exactly the sort of violation it was designed to catch.

That audit is completely unrelated to the original issue filed though, if you find a reproducible URL for the original issue please chime in here again for us to reopen :)

**Comment by Onderhond** [comment]
@patrickhulce hmmm, I don't understand your explanation. 

Putting a link on "NL" would be unnecessary, since it would only be linking to itself. When you are on the English version (https://www.uantwerpen.be/en/) then of course there's a <a href="/nl"> on the NL label. 

https://www.w3.org/TR/2011/WD-html5-20110525/text-level-semantics.html#the-a-element
W3C says this: **If the a element has no href attribute, then the element represents a placeholder for where a link might otherwise have been placed, if it had been relevant.**

That sounds like a perfect match for my use case. Normally it's a link, unless you're on the exact page you're linking to (active state), which makes it irrelevant to link (because it would link to itself). 

I wouldn't mind opening a new ticket for that, unless I'm really misinterpreting something here :)

**Comment by patrickhulce** [comment]
What's valid HTML semantics and what's a best practice from the SEO guidance given to Lighthouse are different things. What you've done is certainly valid HTML, it just doesn't match the intention of the audit which is to help robots learn the structure of the page and its connections. 

When there's an `<a>` without an href, a robot can't guess that it obviously should've linked to itself like we can. The audit is about helping to find those cases, so the structure can be automatically learned. If you don't really care about this concern for this site, then please feel free to ignore the results of the audit here :)

**Comment by Onderhond** [comment]
@patrickhulce understood!

The final thing I'd like to ask then is whether it wouldn't this make more sense as a warning then, rather than an error (since the tool is making assumptions that could be wrong).

I honestly couldn't care less. As a developer I use Lighthouse the way (I assume) it was intended (seeking out improvements and if I don't agree/if they're not relevant, I ignore them). The problem is that the tool is getting traction outside of the developer world, i.e. my boss/clients will be questioning me why they're not getting a full score on SEO. And it'll be pretty hard to explain that the site is perfectly valid, but is getting points deducted for something the tool assumes is wrong, but isn't. And since I could change the code (use a span instead of an href-less a) I will probably be forced to change it in the end, even when it creates more hassle with my css.

I know the tool is not really intended to be a true benchmark, but people out there are starting to use it that way and that's something you guys will eventually have to acknowledge I'm afraid.

**Comment by patrickhulce** [comment]
I'll leave the warning distinction up to the source of truth of the audit @AVGP any thoughts there?

My understanding was that this was in fact an issue from the SEO perspective even if it's valid HTML, so an error in the SEO section is appropriate. But if @AVGP says otherwise we'll happily match whatever the actual SEO impact is.

**Comment by Onderhond** [comment]
Well, this particular use case is about not linking to yourself (which is a best practice as far as I know - but more on the UX/accessibility side of things), but maybe there's an SEO impact I'm not aware of. 

It's not like the crawler misses out on any pages, as he's actually on the page that would otherwise be linked by the href :)

**Comment by AVGP** [comment]
The SEO issue is that without an href with a URL, we can't crawl.

Onderhond <notifications@github.com> schrieb am Di., 1. Dez. 2020, 17:15:

**Comment by Onderhond** [comment]
@AVGP 

I understand that, but why would you need to crawl a page you're already on? That doesn't really makes sense to me. The crawler knows about the page, because he's currently on it.

The link we're talking about is a link that would a href that is equal to the url of the current page (like an active page in a site navigation, or an active page in a language navigation. If the crawler would follow that link, he would land on the exact same page.

**Comment by AVGP** [comment]
Well the check is implemented not for self-referencing links but for those
that are meant to take the crawler elsewhere but doesn't because devs do
something silly.

Onderhond <notifications@github.com> schrieb am Di., 1. Dez. 2020, 17:23:

**Comment by Onderhond** [comment]
I'll point to the W3C link again, because as far as I know, this isn't a "dev doing something silly" (if it is, I'll gladly stop doing it), but a correct and valid implementation of links without a href attribute.

https://www.w3.org/TR/2011/WD-html5-20110525/text-level-semantics.html#the-a-element
W3C says this: **If the a element has no href attribute, then the element represents a placeholder for where a link might otherwise have been placed, if it had been relevant.**

Again, I could replace the a-tag with a span-tag, duplicate some css and make my twig more complex, but that feels like a step backwards to be honest. If my gut feeling is correct, the W3C rule was added specifically to avoid doing all that.

**Comment by AVGP** [comment]
Right, I was paraphrasing.

That use case you're pointing out is valid and fine, but I have seen so
many <a onclick="location.href='/yolo'"> and countless varieties thereof
that we decided to put this in place. And that's an SEO problem if you want
the crawler to follow this.

Onderhond <notifications@github.com> schrieb am Di., 1. Dez. 2020, 17:35:
